be/test/vec/columns/common_column_test.h

```diff
@@ -0,0 +1,766 @@
+        std::vector<size_t> ranges_permutations(ranges_size);
+        std::iota(ranges_permutations.begin(), ranges_permutations.end(),
+                  IColumn::Permutation::value_type(0));
+
+        IColumn::Permutation actual_permutation;
+        IColumn::Permutation expected_permutation;
+
+        for (const auto& range_size : range_sizes) {
+            // step1. generate range field data for column
+            generateRanges(ranges, range_size, columnValueGetter);
+            std::sort(ranges_permutations.begin(), ranges_permutations.end());
+            IColumn::Permutation permutation(ranges_permutations.size());
+            for (size_t i = 0; i < ranges_permutations.size(); ++i) {
+                permutation[i] = ranges_permutations[i];
+            }
+
+            while (true) {
+                // step2. insert range field data into column
+                insertRangesIntoColumn(ranges, permutation, column);
+                static constexpr size_t limit_parts = 4;
+
+                size_t column_size = column.size();
+                size_t column_limit_part = (column_size / limit_parts) + 1;
+
+                for (size_t limit = 0; limit < column_size; limit += column_limit_part) {
+                    assertColumnPermutation(column, true, limit, -1, actual_permutation,
+                                            expected_permutation);
+                    assertColumnPermutation(column, true, limit, 1, actual_permutation,
+                                            expected_permutation);
+
+                    assertColumnPermutation(column, false, limit, -1, actual_permutation,
+                                            expected_permutation);
+                    assertColumnPermutation(column, false, limit, 1, actual_permutation,
+                                            expected_permutation);
+                }
+
+                assertColumnPermutation(column, true, 0, -1, actual_permutation,
+                                        expected_permutation);
+                assertColumnPermutation(column, true, 0, 1, actual_permutation,
+                                        expected_permutation);
+
+                assertColumnPermutation(column, false, 0, -1, actual_permutation,
+                                        expected_permutation);
+                assertColumnPermutation(column, false, 0, 1, actual_permutation,
+                                        expected_permutation);
+
+                if (!std::next_permutation(ranges_permutations.begin(), ranges_permutations.end()))
+                    break;
+            }
+        }
+    }
+
+    // sort calculation: (which used in sort_block )
+    //    get_permutation means sort data in Column as sort order
+    //    limit should be set to limit the sort result
+    //    nan_direction_hint deal with null|NaN value
+    void assertColumnPermutation(const IColumn& column, bool ascending, size_t limit,
```

Comment:
just define actual_permutation and expected_permutation inside this function.

Reply:
done!